tc/core/compiler_api_rfc.h

```diff
@@ -0,0 +1,127 @@
+ *      1 TcExecutor <-> 1 tuple<TC function, input shapes, MappingOptions>
+ *
+ * Compile is templated by the Backend type.
+ * For each backend, the specific Backend type lives in
+ *   hw/hw_backend.h and declares all the required dependent
+ *   **concrete** types.
+ * For example:
+ *   CudaBackend is declared in core/cuda/cuda_backend.h
+ *
+ * struct CudaBackend {
+ *   using ExecutorType = CudaTcExecutor;
+ *   using MappingOptionsType = CudaMappingOptions;
+ *   using ProtobufType = CudaCacheProto;
+ *   using CacheType = CudaCache;
+ *   using CacheEntryType = CudaCacheEntry;
+ *   using CacheKeyType = typename CacheEntryType::Key;
+ *   using CompilationResultType = typename CacheEntryType::Values;
+ *   using RTCFunctionType = CudaRTCFunction;
+ * };
+ *
+ * Sketching usage resembles:
+ *   std::string someTc = "...";
+ *   auto pExecutor = tc::compile<CudaBackend>(
+ *     someTc, tcFunctionName, inputs, mappingOptions);
+ *   auto profilingInfo = pExecutor->run(handle, inputs, outputs, true);
+ *   // alternatively:
+ *   // auto kernelTiming = pExecutor->uncheckedRun(inputs, outputs, true);
+ */
+namespace tc {
+/// Given a TC representation as a TC + TC function name entryPoint, this
+/// function compiles a new TcExecutor for the specified Backend.
+/// For now, contiguous output sizes are inferred given input sizes.
+/// If you need another kernel for another entryPoint or another inputs,
+/// options then just compile another TcExecutor; because atm we fully JIT
+/// specialize on all sizes.
+/// \returns a new TcExecutor on which the run method can be called to run
+/// entryPoint
+template <typename Backend>
+std::unique_ptr<typename Backend::ExecutorType> compile(
+    const std::string& tc,
+    const std::string& entryPoint,
+    const std::vector<const DLConstTensor*>& inputs,
+    /* TODO: in the future also pass outputs for stride and alignment info */
+    const typename Backend::MappingOptionsType& options);
+
+/// Given a TC representation as a TC + TC function name entryPoint and a list
+/// of input tensors that match the definition in the TC function definition
+/// (in positional order), this generates the output TensorInfo resulting from
+/// running inference.
+/// The typical flow is to infer output sizes, allocate/resize them within
+/// you favorite ML framework/tensor library and then call compile and run.
+/// \returns a vector of TensorInfo which can be used for allocating and
+/// performing output shape validation.
+std::vector<TensorInfo> inferOutputTensorInfo(
+    const std::string& tc,
+    const std::string& entryPoint,
+    const std::vector<const DLConstTensor*> inputs);
+
+namespace detail {
+/// Given a TC representation as a TreeRef, this function compiles a new
+/// TcExecutor for the specified Backend.
+/// For now, contiguous output sizes are inferred given input sizes.
+/// If you need another kernel for another TC or another inputs, options then
```

Comment:
Nit: something is wrong in "another inputs, options".
Did you mean "another pair of inputs and options"?
Same phrase also occurs below.